frontend/packages/ux-editor/src/containers/DesignView.tsx

```diff
@@ -37,9 +38,10 @@ export const DesignView = () => {
 
   const { order, containers, components } = layout;
 
-  const addItem = (item: NewDndItem, { parentId, index }: ItemPosition) => {
+  const addItem = async (item: NewDndItem, { parentId, index }: ItemPosition) => {
     const newId = generateComponentId(item.type, layouts);
-    updateFormLayout(addItemOfType(layout, item.type, newId, parentId, index));
+    const container = addItemOfType(layout, item.type, newId, parentId, index);
```

Comment:
Dette kan også gjøres inni hooken. `addItemToLayout` bør ta inn `containerId` og `position` som før, men i stedet for å ta inn en hel `FormComponent`, trenger den bare å ta inn `item.type`. Prøv å skrive den slik at vi kan kalle den på denne måten:
`addItemToLayout({type: item.type, parentId, index})`

Reply:
Det gjennomgående prinsippet her er at React-komponenter skal være så dumme som mulig. De skal kun presentere informasjon og trigge kall til andre funksjoner som tar seg av logikken. Dette er viktig for å ha en klar fordeling av hvilke oppgaver de forskjellige delene av applikasjonen skal ha. Nå vet jeg at dette ikke er så godt fulgt opp overalt i Studio, men det er noe av det vi jobber med å rydde opp i 🙂